### ISSUE TITLE
Wrong base URL with nuxi build with NUXT_PUBLIC_SITE_URL env var set

### ISSUE THREAD

**Issue by learntheropes** [harlan-zw/nuxt-simple-sitemap#104]
### Describe the bug

I have `NUXT_PUBLIC_SITE_URL=http://localhost:3000`

When I run `nuxi dev` the base URL in the sitemap is set correctly to `http://localhost:3000`.

However, when I run `nuxi build && nuxi start` the base URL is set to `https://false`

I have a reproduction but is not relevant since on stackblitz you can't set env vars, and you can't run `nuxi build`

### Reproduction

https://stackblitz.com/edit/nuxt-sitemap?file=package.json

### System / Nuxt Info

```Shell
- Operating System: Linux
- Node Version:     v18.17.0
- Nuxt Version:     3.6.5
- Nitro Version:    2.5.2
- Package Manager:  npm@9.6.7
- Builder:          vite
- User Config:      runtimeConfig, app, $production, css, components, modules, robots, sitemap, i18n, content, auth, pinia, imports, image, nitro, delayHydration
- Runtime Modules:  nuxt-simple-robots@3.0.3, nuxt-simple-sitemap@3.1.0, @nuxtjs/i18n@8.0.0-beta.13, @nuxt/content@2.7.1, nuxt-content-git@3.0.32, @sidebase/nuxt-auth@0.6.0-beta.3, @pinia/nuxt@0.4.11, @nuxt/image@1.0.0-rc.1, nuxt-delay-hydration@1.2.1
- Build Modules:    -
```

**Comment by learntheropes** [comment]
```js
export default defineNuxtConfig({
  site: {
    url: process.env,NUXT_PUBLIC_SITE_URL
  }
})
```
Fixes the issue, but this should be not needed, at least accordingly to the documentation.

**Comment by harlan-zw** [comment]
Hey @learntheropes, thanks for the issue.

This should be fixed in [3.1.3](https://github.com/harlan-zw/nuxt-simple-sitemap/releases/tag/v3.1.3), let me know if you have any issues.

**Comment by learntheropes** [comment]
I confirm that is resolved.

Thank you for the quick fix!
